sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/UseUriInsteadOfString.cs

```diff
@@ -60,5 +62,22 @@ class Program
         string urifoo { get; set; } // Compliant
         string urlfoo { get; set; } // Compliant
         string urnfoo { get; set; } // Compliant
+
+        string ParamTest(string uri) => ""; // Noncompliant {{Either change this parameter type to 'System.Uri' or provide an overload which takes a 'System.Uri' parameter.}}
+//                       ^^^^^^
+        string ParamTest2(int url) => ""; // Compliant
+        string ParamTestOverload(string uriParam) => ""; // Noncompliant {{Either change this parameter type to 'System.Uri' or provide an overload which takes a 'System.Uri' parameter.}}
+//                               ^^^^^^
+        string ParamTestOverload(int uriParam) => ""; // Compliant
+        string MultipleParams(string uriParam, object urnParam, string urlParam, string andThisIsFine) => "";
+//                            ^^^^^^ Noncompliant {{Either change this parameter type to 'System.Uri' or provide an overload which takes a 'System.Uri' parameter.}}
+//                                                              ^^^^^^ Noncompliant@-1 {{Either change this parameter type to 'System.Uri' or provide an overload which takes a 'System.Uri' parameter.}}
```

Comment:
Could you put each rule test into a different class (even if it is on the same file) just so that it is easy to understand which rule covers this test? It is to ease readability when we will get back to this in a couple of months.